sdk/keyvault/keyvault-keys/src/keysModels.ts

```diff
@@ -358,6 +358,17 @@ export interface CreateRsaKeyOptions extends CreateKeyOptions {
   publicExponent?: number;
 }
 
+/**
+ * An interface representing the optional parameters that can be
+ * passed to {@link createOctKey}
+ */
+export interface CreateOctKeyOptions extends CreateKeyOptions {
+  /**
+   * Whether to create a hardware-protected key in a hardware security module (HSM).
+   */
+  hsm?: boolean;
```

Comment:
I went with `hsm` because we already use `hsm` on `createEcKeyOptions` and `createRsaKeyOptions` - but feel free to let me know if we should break tradition here to use `hardwareProtected`